### PR TITLE
feat(test-optimization): create final_status tag on test event for playwright

### PR DESCRIPTION
**Pull request by cbasitodx**
### What does this PR do?
Create a `@test.final_status` tag that is added to every playwright test event that represents the final execution of that test (there can be multiple executions of a single test due to retry mechanisms being active)

### Motivation
Tracers send each test execution as a span to the backend. When retry mechanisms are enabled a single test can run several times with different outcomes. These mechanisms differ in behavior. Some stop once the test reaches the desired state such as ATR ending when a test passes. Others always run a fixed number of attempts such as EFD or Attempt to fix.

To ensure the correct final result for the customer some intermediate outcomes are suppressed and the testing framework receives an adjusted final status. For example if an early EFD attempt passes but later attempts fail the test is still reported as passed and the failing attempts are hidden to avoid failing CI pipelines.

Both external and internal users have requested a way to query tests in Datadog by this final status. Their main goal is to build monitors and alerts for hard failures on default branches.

To support this a new tag will be added to the last retry of a test named: `@test.final_status:pass|fail|skip`

### CODE REVIEW

File: packages/datadog-instrumentations/src/playwright.js [hunk 1]
```diff
@@ -613,6 +654,24 @@ function dispatcherHookNew (dispatcherExport, runWrapper) {
...
+      // main process, so workers only need to act on the _ddIsFinalExecution flag.
+      let isFinalExecution
+      if ((test._ddIsNew || test._ddIsModified) && !test._ddIsAttemptToFix && isEarlyFlakeDetectionEnabled) {
+        const testFqn = getTestFullyQualifiedName(test)
+        const efdTestStatuses = testsToTestStatuses.get(testFqn) || []
+        isFinalExecution = efdTestStatuses.length === earlyFlakeDetectionNumRetries + 1
+      } else if (test._ddIsAttemptToFix) {
+        isFinalExecution = !!(test._ddHasPassedAttemptToFixRetries || test._ddHasFailedAttemptToFixRetries)
+      } else {
+        isFinalExecution = !testWillRetry(test, testStatus)
```
[the remark this answers is not in the document]
Comment (cbasitodx) [reply packages/datadog-instrumentations/src/playwright.js:700]: Fixed in a separate PR (https://github.com/DataDog/dd-trace-js/pull/8249)

File: packages/datadog-instrumentations/src/playwright.js [hunk 2]
```diff
@@ -293,6 +293,32 @@ function testWillRetry (test, testStatus) {
...
+  hasFailedAttemptToFixRetries,
+  testStatus,
+}) {
+  if (!isFinalExecution) {
+    return
+  }
+  if (isDisabled || isQuarantined || testStatus === 'skip') {
+    return 'skip'
+  }
+  if (isAtrRetry || isEfdRetry) {
```
Comment (juan-fernandez) [inline packages/datadog-instrumentations/src/playwright.js]: sneaky edge case here: 
1. EFD runs original + 3 clones for a new test.
2. One clone passes, but the original run fails and happens to finish last.
3. `isFinalExecution=true` for the original because it is the 4th reported status.
4. `isEfdRetry=false`, so `getFinalStatus()` returns raw 'fail'.
5. Expected EFD final status should be 'pass', because not all EFD executions failed.

A practical fix is to pass an explicit “EFD-managed test” boolean into `getFinalStatus`, instead of using only `isEfdRetry`:

```javascript
  const isEfdManagedTest = (test._ddIsNew || test._ddIsModified) &&
    !test._ddIsAttemptToFix &&
    isEarlyFlakeDetectionEnabled
```

and then 
```suggestion
  if (isAtrRetry || isEfdManagedTest) {
```
instead of using `isEfdRetry`

The value needs to be send via `ddProperties`
```javascript
  _ddIsEfdManagedTest: isEfdManagedTest
```
and then:
```javascript
  isEfdManagedTest: test._ddIsEfdManagedTest
```

**Review by juan-fernandez — COMMENTED** [review]
great job! Only one sneaky edge case uncovered

File: packages/datadog-instrumentations/src/playwright.js (hunk 2 above)
Comment (cbasitodx) [reply packages/datadog-instrumentations/src/playwright.js]: awesome catch! 😮

**Review by juan-fernandez — APPROVED** [review]
great job!
